gateway-ha/src/main/java/io/trino/gateway/ha/router/RoutingRulesManager.java

```
@@ -0,0 +1,84 @@
+            YAMLParser parser = new YAMLFactory().createParser(content);
+            List<RoutingRules> routingRulesList = new ArrayList<>();
+            while (parser.nextToken() != null) {
+                RoutingRules routingRules = yamlReader.readValue(parser, RoutingRules.class);
+                routingRulesList.add(routingRules);
+            }
+            return routingRulesList;
+        }
+        catch (IOException e) {
+            throw new IOException(e);
+        }
+    }
+
+    public List<RoutingRules> updateRoutingRules(RoutingRules routingRules, RoutingRulesConfiguration configuration, ObjectMapper yamlReader)
+            throws IOException
+    {
+        String rulesConfigPath = configuration.getRulesConfigPath();
+        List<RoutingRules> routingRulesList = new ArrayList<>();
+        try {
+            String content = new String(Files.readAllBytes(Paths.get(rulesConfigPath)), UTF_8);
+            YAMLParser parser = new YAMLFactory().createParser(content);
+            while (parser.nextToken() != null) {
+                RoutingRules routingRule = yamlReader.readValue(parser, RoutingRules.class);
+                routingRulesList.add(routingRule);
+            }
+
+            for (int i = 0; i < routingRulesList.size(); i++) {
+                if (routingRulesList.get(i).name().equals(routingRules.name())) {
+                    routingRulesList.set(i, routingRules);
+                    break;
+                }
+            }
+
+            ObjectMapper yamlWriter = new ObjectMapper(new YAMLFactory());
+            StringBuilder yamlContent = new StringBuilder();
+            for (RoutingRules rule : routingRulesList) {
+                yamlContent.append(yamlWriter.writeValueAsString(rule));
+            }
+            Files.write(Paths.get(rulesConfigPath), yamlContent.toString().getBytes(UTF_8));
+        }
+        catch (IOException e) {
+            throw new IOException(e);
```

Please rethrow an appropriate exception.